src/dma/mod.rs

```diff
@@ -1059,6 +1059,11 @@ where
         }
     }
 
+    /// Access the owned peripheral for reading
+    pub fn peripheral(&self) -> &PERIPHERAL {
```

Comment:
DMA code for this HAL is not mine, so I need to clarify is such access safe?

Reply:
cc @thalesfragoso

Reply:
@gernoteger I still can't merge this as this method gives unsafe access any DMA payload.
Possible solution:
https://github.com/stm32-rs/stm32f4xx-hal/commit/b2752339a04cc631e37272e21bdcf5429532be25

Reply:
thanks for doing all the work. I incorporated your suggestion nearly 1:1, only fixed a small typo in a comment